src/main/python/ttconv/imsc/imsc_reader.py

```diff
@@ -476,90 +530,353 @@ def process(context, inherited_space, inherited_lang, ttml_element):
 
           element.push_child(child_element)
 
-      if ttml_child_element.tail is not None:
+      # process tail text node
 
-        element.push_child(model.Text(context.doc, ttml_element.text))
+      if ttml_child_element.tail:
+        element.push_child(model.Text(context.doc, ttml_child_element.tail))
 
     return element
 
-class BrElement:
-  '''Process the TTML <br> element
+
+class RubyElement:
```

Comment:
In order to limit the size of the files, shouldn't the Ruby-related class definitions be placed into a specific file? And it would be easier to review :wink:

Reply:
I think all element processing should be either kept in the same file (since they are all so similar and dependent) or create one file per element -- there is nothing really special about ruby element processing other than, by habit, I worked on them last :) 
There *is* a lot of code duplication across element processing. Perhaps it is possible to reduce it. Processing is however both annoyingly similar and but also slightly different across elements.

Reply:
OK then, we'll see along the project progress if we can simplify all this!